File: MediaPlayer/MediaPlayer/Util/Category/UIImageView+Extention.m
//
//  UIImageView+Extention.m
//  StudentLive
//
//  Created by hefanghui on 2017/8/4.
//  Copyright © 2017年 hqyxedu. All rights reserved.
//
//  Main function:UIImageView拓展类
//
//  Other specifications:包含：根据图片名称创建实例并sizeToFit的快速构造方法
//
//  ************************************************************************

#import "UIImageView+Extention.h"

@implementation UIImageView (Extention)

+ (instancetype)imageNamed:(NSString *)imageName {
    UIImageView *imageView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:imageName]];
    [imageView sizeToFit];
    return imageView;
}

@end

File: MediaPlayer/MediaPlayer/Util/Category/UIImageView+Extention.swift
//
//  UIImageView+Extention.swift
//  USchool
//
//  Created by hefanghui on 2017/9/14.
//  Copyright © 2017年 topglobaledu. All rights reserved.
//
//  Main function:UIImageView便利构造初始化方法
//
//  Other specifications:
//
//  ************************************************************************

import UIKit

extension UIImageView {
    class func imageView(imageName: String) -> UIImageView {
        let imageView: UIImageView = UIImageView.init(image: UIImage.init(named: imageName));
        return imageView;
    }
    
    class func imageView(color: UIColor) -> UIImageView {
        let imageView: UIImageView = UIImageView.init();
        imageView.image = self.p_imageWithColor(color: color)
        return imageView;
    }
    
    private class func p_imageWithColor(color: UIColor) -> UIImage {
        let rect: CGRect = CGRect.init(x: 0.0, y: 0.0, width: 1.0, height: 1.0)
        UIGraphicsBeginImageContext(rect.size)
        let context = UIGraphicsGetCurrentContext()
        context!.setFillColor(color.cgColor)
        context!.fill(rect)
        let image: UIImage = UIGraphicsGetImageFromCurrentImageContext()!
        UIGraphicsEndImageContext();
        
        return image;
    }
}
